components/github-api-client/src/metrics.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Centralized definition of all Github API client metrics that we
+//! wish to track.
+
+use std::borrow::Cow;
+use builder_core::metrics;
+
+pub type Endpoint = &'static str;
```

Comment:
As a design choice the original code decided _not_ to allow names to come in as arbitrary strings.  Your use case makes sense here... we could think of having enums as well to have a bit more control over the strings.